vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
-    mcRightSettings.ForwardSpeedLimit = limit;
-
+    mcSettings.ForwardSpeedLimit = limit;
     return HAL_OK;
 }
 
 HAL_StatusTypeDef setTorqueLimit(float limit)
 {
-    mcRightSettings.DriveTorqueLimit = limit;
-    mcLeftSettings.DriveTorqueLimit = limit;
-
+    mcSettings.DriveTorqueLimit = limit;
     return HAL_OK;
 }
 
 HAL_StatusTypeDef setDischargeCurrentLimit(float limit)
 {
-    mcRightSettings.DischargeCurrentLimit = limit;
-    mcLeftSettings.DischargeCurrentLimit = limit;
+    mcSettings.DischargeCurrentLimit = limit;
 	return HAL_OK;
 }
 
-// TODO: Probably need to set speed limits after init
-HAL_StatusTypeDef mcInit()
-{
-	DEBUG_PRINT("Discharging MCs for 1 second before turning on\n");
-
-    if (mcRightCommand(INVERTER_DISCHARGE_DC_LINK) != HAL_OK) {
-        ERROR_PRINT("Failed to send discharge command to MC Right");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef mcReadParamCommand(uint16_t address, uint16_t data) {
+    VCU_INV_Parameter_RW_Command = INVERTER_PARAM_READ;
+    VCU_INV_Parameter_Address = address;
+    VCU_INV_Parameter_Data = data; // Is this needed for reading? todo
 
-    if (mcLeftCommand(INVERTER_DISCHARGE_DC_LINK) != HAL_OK) {
-        ERROR_PRINT("Failed to send discharge command to MC Left");
+    if (sendCAN_MC_Read_Write_Param_Command() != HAL_OK) {
+        ERROR_PRINT("Failed to send read param message to MC\n");
         return HAL_ERROR;
     }
+    return HAL_OK;
+}
 
-    vTaskDelay(pdMS_TO_TICKS(MC_INIT_DISCHARGE_TIME_MS));
+HAL_StatusTypeDef mcWriteParamCommand(uint16_t address, uint16_t data) {
+    VCU_INV_Parameter_RW_Command = INVERTER_PARAM_WRITE;
+    VCU_INV_Parameter_Address = address;
+    VCU_INV_Parameter_Data = data;
 
-    if (mcRightCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Right");
+    if (sendCAN_MC_Read_Write_Param_Command() != HAL_OK) {
+        ERROR_PRINT("Failed to send write param message to MC\n");
         return HAL_ERROR;
     }
+    return HAL_OK;
+}
 
-    if (mcLeftCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Left");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef mcClearFaults() {
+    return mcWriteParamCommand(INVERTER_FAULT_CLEAR_ADDRESS, 0);
+}
 
+HAL_StatusTypeDef mcInit() {
     DEBUG_PRINT("Waiting for MC to complete startup checks\n");
+    // Todo - Read and check EEPROM values
 
-    // TODO: Do we need to check the voltage values from the MC, or are the BMU
-    // startup checks sufficient?
-
-    DEBUG_PRINT("Starting MC Left...\n");
+    DEBUG_PRINT("Starting MC ..\n");
 
     uint32_t startTick = xTaskGetTickCount();
-    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) &&
-           ((StateInverterLeft & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED))
+
+    // Attempt to disable the lockout
+    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) && 
```

Comment:
Have you tried lowering the timeout to see how close we are to the real limit. I say real limit bc sometimes datasheet values are not always the same as experimental results